packages/protocol/test/customHooks.ts

```diff
@@ -0,0 +1,29 @@
+import { sleep } from '../lib/test-utils'
+
+// Handles flakey `error: Invalid JSON RPC response: ""` error that seems to be caused by port exhaustion in CI.
+// See https://github.com/ethereum/web3.js/issues/3425 and https://github.com/ethereum/web3.js/issues/926.
+export const beforeEachWithRetries = (
```

Comment:
What do you think about using the function we already have in utils? https://github.com/celo-org/celo-monorepo/blob/master/packages/utils/src/async.ts#L12

Reply:
+1

Reply:
Great idea!